### ISSUE TITLE
[Bug] I used undetected-playwright but always get detected as bot (robot)

### ISSUE THREAD

**Issue by thakkerurvish** [QIN2DIM/undetected-playwright#11]
Context:

I tried to leverage undetected-playwright to safe guard my browser instance to be detected as bot by the websites. However, with all the websites I worked with it showed no benefit over normal playwright. I always run into bot detected using undetected-playwright as well same as it would happen without using undetected version. Any thoughts, on addressing this issue in the future or was this just a one time project? Just Curious about it.

Kudos on the great attempt, but it seems the current version of evasions has already been escaped by the modern day applications (websites).

Some everyday example websites, where it does not work:

->https://www.etsy.com/
->https://www.reuters.com/
->https://www.viator.com/
->https://www.petco.com/

**Comment by princewillingoo** [comment]
Hi, @thakkerurvish
Did you find any fix?

**Comment by zero-stroke** [comment]
Is this in instances in which undetected selenium does work in remaining undetected?

**Comment by princewillingoo** [comment]
> Is this in instances in which undetected selenium does work in remaining undetected?

Try using - https://pypi.org/project/playwright-stealth/

**Comment by gexiea** [comment]
Yeah, playwright-stealth is pretty much dead for strict sites like Etsy or Reuters. They check the actual binary signatures now, not just JS properties.

I got tired of patching Chromium myself. Lately, I've just been pulling the docker image from stealthbrowser.org. It has noVNC built-in which is nice for visual debugging. The annoying part is their business model—they charge per custom fingerprint profile you generate.

To get around paying for it, I just use their base browser. Since they already stripped out the low-level WebDriver/CDP leaks in their custom Chromium, captchas naturally don't flag you right away. Then, instead of paying for their native fingerprint generator, I just use standard JS injection to rotate the canvas/WebGL/UA stuff myself.

Obviously, JS spoofing isn't as bulletproof as their paid native kernel-level fingerprints, but it's good enough to scrape Reuters and Viator without getting blocked, and it saves a ton of money.
